user_acceptance_tests/tests/mysql.rb

```diff
@@ -40,6 +40,34 @@
     expect(response.code).must_equal(200)
   end
 
+  it 'X-DEMO-INVALID-QUERY-PRE behavior should throw on GET /api/inventory' do
+    headers = { 'X-DEMO-INVALID-QUERY-PRE' => 0 }
+    err = _(-> { RestClient.get("#{service_url}/api/inventory", headers) }).must_raise(RestClient::ExceptionWithResponse)
+    expect(err.http_code).must_equal(500)
+    expect(err.response).must_include('ER_NO_SUCH_TABLE')
+  end
+
+  it 'X-DEMO-INVALID-QUERY-POST behavior should throw on GET /api/inventory' do
+    headers = { 'X-DEMO-INVALID-QUERY-POST' => 0 }
+    err = _(-> { RestClient.get("#{service_url}/api/inventory", headers) }).must_raise(RestClient::ExceptionWithResponse)
+    expect(err.http_code).must_equal(500)
+    expect(err.response).must_include('ER_NO_SUCH_TABLE')
+  end
+
+  it 'X-DEMO-INVALID-QUERY-PRE behavior should throw on GET /api/inventory/1' do
+    headers = { 'X-DEMO-INVALID-QUERY-PRE' => 0 }
+    err = _(-> { RestClient.get("#{service_url}/api/inventory/1", headers) }).must_raise(RestClient::ExceptionWithResponse)
+    expect(err.http_code).must_equal(500)
+    expect(err.response).must_include('ER_NO_SUCH_TABLE')
+  end
+
+  it 'X-DEMO-INVALID-QUERY-POST behavior should throw on GET /api/inventory/1' do
+    headers = { 'X-DEMO-INVALID-QUERY-POST' => 0 }
+    err = _(-> { RestClient.get("#{service_url}/api/inventory/1", headers) }).must_raise(RestClient::ExceptionWithResponse)
+    expect(err.http_code).must_equal(500)
+    expect(err.response).must_include('ER_NO_SUCH_TABLE')
+  end
+
```

Comment:
these look quite reasonable. ill copy this setup. i was thinking this would also be possible as a unit test or set of tests (which i think are better to have vs more UATs), but given how simple this looks, I'm willing to just stick with this.